### DISCUSSION TITLE
BadMonitoredItemFilterInvalid

### DISCUSSION THREAD

**Discussion by ts4iot** [FreeOpcUa/python-opcua#1499] [Q&A] [answered]
I try to subscribe to Alarms and Events on a Siemens WinCC.

I use the following example as basis:
[client-events.py](https://github.com/FreeOpcUa/python-opcua/blob/master/examples/client-events.py)

I adjusted the main funktion at three points:

- client = Client("opc.tcp://172.31.73.22:4862")
- obj = root.get_child(["0:Objects", "1:WinCC", "1:@LOCALMACHINE::"])
- myevent = root.get_child(["0:Types", "0:ObjectTypes", "0:BaseObjectType", "0:BaseEventType", "2:WinCCEventType"])

**After running the example I got the following result:**

Requested session timeout to be 3600000ms, got 300000ms instead
Objects node is:  i=84
MyObject is:  ns=1;s=f|@LOCALMACHINE::
MyFirstEventType is:  i=2130
ServiceFault from server received while waiting for publish response
Traceback (most recent call last):
  File "03_subscribe_event_.py", line 51, in <module>
    handle = sub.subscribe_events(obj, myevent)
  File "/home/deviceadmin/.local/lib/python3.7/site-packages/opcua/common/subscription.py", line 216, in subscribe_events
    return self._subscribe(sourcenode, ua.AttributeIds.EventNotifier, evfilter, queuesize=queuesize)
  File "/home/deviceadmin/.local/lib/python3.7/site-packages/opcua/common/subscription.py", line 234, in _subscribe
    mids[0].check()
  File "/home/deviceadmin/.local/lib/python3.7/site-packages/opcua/ua/uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadMonitoredItemFilterInvalid: "The monitored item filter parameter is not valid."(BadMonitoredItemFilterInvalid)

Can someone help me to set the filter parameter correctly. Currently I have the filter on the default setting.

**Comment by schroeder-** [comment, marked as answer]
This library is not longer supported see #1475. If the error exists in asyncua, open a issue there.
